packages/vhd-lib/createStreamNbd.js

```diff
@@ -69,10 +69,33 @@ exports.createNbdVhdStream = async function createVhdStream(nbdClient, sourceStr
     }
   }
 
+  const totalLength = (offsetSector + blockSizeInSectors + 1) /* end footer */ * SECTOR_SIZE
+  let readedLength = 0
+  let lastProgress = 0
+  let lastUpdate = 0
+  function* trackAndYield(buffer) {
+    readedLength += buffer.length
+    const currentProgress = readedLength / totalLength
+    const formattedProgress = currentProgress.toFixed(2)
+    const now = Date.now()
+    const diffInSeconds = (now - lastUpdate) / 1000
+
+    if (lastProgress !== formattedProgress) {
+      lastProgress = formattedProgress
+      lastUpdate = now
+      stream.emit('progress', lastProgress)
+    } else if (diffInSeconds > 5) {
```

Comment:
5 should be configurable ( passed as en option of this method), with a default value set in a constant

Also if a block is transferred at the 4.9s mark,  there won't be any progress before the next block 

I I think I would do something like that  ( but with better variable and function naming if possible) 

```js
async function createNBD(, , {maxDurationBetweenProgress = MAX_DURATION_BETWEEN_PROGRESS, minProgressThreshold = MIN_PROGRESS_THRESHOLD} = {})

function emitThrottle(){
    // changed enough or changed a little since a long time 
    // try to always compare int value, float equality may be tricky 
    
     if(readLength - last_read_length > minProgressThreshold * totalLength
        ||  (new Date() - last_emit > maxDurationBetweenProgress && readLength !== last_read_length)){
         stream.emit('progress', progress)
          last_read_length = readLength
          last_emit = new Date()
     }
}
cosnt interval = setInterval(()=>emitThrottle(progress), maxDurationBetweenProgress)
function * trackAndYield(buffer){
   ...
   emitThrottle(progress)
   ...
}

stream.on('end', ()=>clearInterval(interval))
stream.on('error', ()=>clearInterval(interval))

Reply:
Why it should be configurable, if is not configurable directly from the `XO config`?

Reply:
Almost all the magic  values are finally changed on a custom user installation during a support. Exposing this in the function prototype is almost free, and will allow more flexibility later. At least it should be a constant with an explicit name, and eventually a little explication in a comment (like is it in second, millisecond, percent, byte, ... )